### ISSUE TITLE
Please complete total candidate, entry and pass counts

### ISSUE THREAD

**Issue by aba-sah** [aba-sah/sta-it402-dresscode#3]
@matapaska, @mshedededen (@BlagaBaycheva) - please complete the [total candidate, entry and pass counts](https://github.com/aba-sah/sta-it402-dresscode/blob/36d4223b094ec326324cc1ef69e4c891f4035c05/code/data/candidate_counts.xlsx) for the qualifications you're working with - I found they're in either the last or close to the last sheet in each workbook - that will allow us to normalise trends across years.

I've left it all as three sheets in an xlsx file for convenience, will convert each sheet to csv once complete. The headers map to the qualifications file, so if you add to the latter please create entries at the tail for the former.

Please include a note when you commit a new version so I pick up all entries, on the off chance commits aren't sequential.

**Comment by mshedededen** [comment]
Got a few questions about this:
1. What is the difference between 'total_candidates' and 'total_subject_entries'?
2. How did you do this, without trapsing through every Excel file and counting? Seems like a lot of work.
3. What is the purpose of this? I know you say it's to "normalise trends", but I don't particularly understand still why.

**Comment by aba-sah** [comment]
@mshedededen - as in my original note - the information is in the last or one of the trailing sheets - I've attached below a snapshot of the 2005 higher sheet as an example. 

So, if in one year there are 10K candidates (student counts - the actual term varies, but the meaning can be inferred, candidates is the most meaningful/representative of them) and in the following year 8.5 and the next 9.6, but the same number of subject entries for, e.g., CS, across all three years you cannot directly compare the entries. If you look simply at counts it would appear that there are no changes across the years. So you must normalise subject entries across candidates and the other two variables so that the trends are comparable and show actual change over time. Percentages do this to a point but do not provide as much detail. 

FYI - you can easily calculate total subject entries, but there is no other way to determine candidate counts, and calculating passes is problematic as the level of detail varies. But all three values are key. The following is a good way to double-check you're picking up the correct information:
1. (Subject) entries > candidates (in theory >=, but in practice you will never find that each student takes exactly one subject)
2. Subject entries >= total passes (per subject) 
            => total entries >= total passes (all subjects)



![it-402-candidate_subject_and_pass_summaries](https://user-images.githubusercontent.com/3797671/111459721-fab10b80-8712-11eb-8b31-edbb51d14e8a.png)

**Comment by mshedededen** [comment]
Right, I see a point for more recent data, since the reporting structure changed. But, is this necessary for earlier years (e.g. pre-2002) since there is no distinction between entries and candidates ?

**Comment by aba-sah** [comment]
Yes. All three variables change over time. So capturing only some of the data presents at best a partial picture and in practice really an incorrect picture, as it would simply introduce inconsistency. It's not about the reporting structure but data we need. FYI, there IS a distinction between subject entries and candidates - which I explain (look again at point 1 above) - you can see this in the snapshot as well.

It is a tedious exercise, as for all data processing, but on balance not really worth including only part of it; It would actually be better to exclude it completely (not a good idea either - it's actually come up a couple of times, and noted not having it as a problem). A good way to break the tedium is to split the task - do, say, 5 years at a go a couple times a day. Or if you're going to spend another n sessions on pre-processing, say, split doing it across them.

**Comment by mshedededen** [comment]
I don't understand so i'll ask tomorrow.
